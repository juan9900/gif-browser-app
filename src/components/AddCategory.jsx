import {useState} from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({onNewCategory}) => {
    const [inputValue, setInputValue] = useState('');
    
    const onInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const onSubmitForm = (e) => {
        e.preventDefault();
        const searchTerm = (inputValue.charAt(0).toUpperCase() + inputValue.substring(1).toLowerCase()).trim();
        if(searchTerm.length <= 1) return;
        onNewCategory(searchTerm);
        setInputValue('');

    }
    return (
        <>
            <form aria-label='form' onSubmit={onSubmitForm}>
                <input type="text" 
                    placeholder='Search for Gifs'
                    onChange={onInputChange}
                    value={inputValue}/>
            </form>
        </>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}