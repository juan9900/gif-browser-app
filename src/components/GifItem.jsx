import PropTypes from 'prop-types';

export const GifItem = ({title, url, shareLink}) => {
  return (
    <div className="gif">
      <div className="gif-container">
      {/* <div className="gif-link-container">
        <div>
            <h4>Share</h4>
            <div className="gif-share-container">
            <a href={url} target="_blank"><i className="fa-brands fa-whatsapp"></i></a>
            <a href={url} target="_blank"><i className="fa-brands fa-twitter"></i></a>
            <a href={`https://twitter.com/intent/tweet/text?=${shareLink + '  via @Giphy'}`} target="_blank"><i className="fa-solid fa-download"></i></a>
          </div>
        </div>
          
      </div> */}
      <img src={url} alt={title}/>
      </div>
      
      <h3>{title}</h3>
    </div>
  )
}


GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  shareLink: PropTypes.string
  
}