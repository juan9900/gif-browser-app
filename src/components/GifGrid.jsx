// import { useEffect, useState } from "react"
// import { getGifs } from "../helpers/getGifs";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";

export const GifGrid = ({category}) => {

    const {gifList, isLoading} = useFetchGifs(category);
    

    return (
        
        <div>
            {isLoading ? <h2>Cargando...</h2> : (
                <>
            <h2>{category}</h2>
            
            <ul className="gif-topic-container">
            {gifList.length > 0 ?
                // <p>{gifList[0].url}</p>
                // <img src={gifList[0].url} alt="" srcset="" />
                gifList.map(gif =>  (
                        <GifItem key={gif.id} 
                        // title={gif.title} url={gif.url}
                        { ...gif } //<-- Esto es mejor
                        />
                    )
                )
                : <p>No se encontaron resultados.</p>
            }
            </ul>
            </>
            )}
        </div>
    )
}
