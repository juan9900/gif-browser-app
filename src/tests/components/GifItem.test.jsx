import {render, screen} from '@testing-library/react'
import { GifItem } from '../../components';
describe('Tests for <GifItem />', () => { 
    const title = 'One Punch';
    const url = 'http://one-punch.com/saitama.jpg';
    const shareLink = 'http://one-punch.giphy.com/saitama.jpg';
    test('should return a snapshot', () => { 
        const {container} = render(<GifItem title={title} url={url} shareLink={shareLink}/>);
        expect(container).toMatchSnapshot();
     });

     test('should display the image with the url as src and title as alt', () => { 
        render(<GifItem title={title} url={url} shareLink={shareLink}/>);
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);
        // Esta manera es mas limpia y mas elegante que la de arriba:
        const {src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('should display the title of the gif inside an H3', () => { 
        render(<GifItem title={title} url={url} shareLink={shareLink} />);
        expect(screen.getByRole('heading',{level: 3}).innerHTML).toBe(title);
     });
 })