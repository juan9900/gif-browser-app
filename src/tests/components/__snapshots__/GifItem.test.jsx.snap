// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <GifItem /> should return a snapshot 1`] = `
<div>
  <div
    class="gif"
  >
    <div
      class="gif-container"
    >
      <div
        class="gif-link-container"
      >
        <div>
          <h4>
            Share
          </h4>
          <div
            class="gif-share-container"
          >
            <a
              href="http://one-punch.com/saitama.jpg"
              target="_blank"
            >
              <i
                class="fa-brands fa-whatsapp"
              />
            </a>
            <a
              href="http://one-punch.com/saitama.jpg"
              target="_blank"
            >
              <i
                class="fa-brands fa-twitter"
              />
            </a>
            <a
              href="https://twitter.com/intent/tweet/text?=http://one-punch.giphy.com/saitama.jpg  via @Giphy"
              target="_blank"
            >
              <i
                class="fa-solid fa-download"
              />
            </a>
          </div>
        </div>
      </div>
      <img
        alt="One Punch"
        src="http://one-punch.com/saitama.jpg"
      />
    </div>
    <h3>
      One Punch
    </h3>
  </div>
</div>
`;
